### PR TITLE
Validation specs

### PR DESCRIPTION
**Pull request by Busiwa24**
**In the following PR, I have completed the following tasks:**
- Added the following validations:

For the User model:
- Ensured that the name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

For the Post model:
- Ensured that the title is not blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Added unit specs for all of your models' methods and validations.
- Added flash messages in the create actions in all my controllers.
- Ensured that the N+1 problem is solved when fetching all posts and their comments for a user according to specified guidelines for the Blog Application

### CODE REVIEW

File: db/migrate/20211202105020_create_users.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ def change
       t.string :name
```
Comment (SajjadAhmad14) [inline db/migrate/20211202105020_create_users.rb]: There are some double migrations in the database which are causing migration errors. Kindly try running` rails db: migrate` and remove all the duplicate migrations.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -1,5 +1,47 @@
 require 'rails_helper'
 
 RSpec.describe Post, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'Post Model Validations' do
+    user = User.create(name: 'Dammy', bio: 'Dammy is so cool!')
+    subject do
+      Post.new(title: 'Bootstrap vs Tailwind', text: 'Let\'s  talk about Bootstrap and Tailwind', author_id: user)
+    end
```
Comment (SajjadAhmad14) [inline spec/models/post_spec.rb]: Please add a unit test for `#update_posts_counter` of Post model as it's required to test custom model methods.

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -1,5 +1,19 @@
 require 'rails_helper'
 
 RSpec.describe Like, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'Comment Likes Validations' do
+    subject { Like.new(author_id: 1, post_id: 1) }
+
+    before { subject.save }
+
```
Comment (SajjadAhmad14) [inline spec/models/like_spec.rb]: Please add a unit test for `#update_likes_counter` of the Like model as it's required to test custom model methods.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Busi, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/models/like_spec.rb [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
+  before { subject.save }
+
+  it 'User ID should be a number' do
+    subject.user_id = 'pizza'
+    expect(subject).to_not be_valid
+  end
+
+  it 'Post ID should be a number' do
+    subject.post_id = 'pasta'
+    expect(subject).to_not be_valid
```
Comment (Dipeshtwis) [inline spec/models/like_spec.rb]: - [ ] There are 20 tests that are failing. Here you are using `user_id`, but you have to change this to `author_id`. Please fix it.

File: spec/models/post_spec.rb [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+  before { subject.save }
+
+  it 'User ID should be a number' do
+    subject.user_id = 'pizza'
+    expect(subject).to_not be_valid
+  end
+
+  it 'Post ID should be a number' do
+    subject.post_id = 'pasta'
+    expect(subject).to_not be_valid
```
Comment (Dipeshtwis) [inline spec/models/post_spec.rb]: - [ ] Here as well you are using `user_id`, but you have to change this to `author_id`. Please fix it.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Busiwa24, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️

- [ ] There are some tests that are failing, Please fix them before asking for a code review. Below I have pointed out some of them why this is happening.

- [ ] Also I would recommend mentioning the command on how to test in the README file. `bundle exec rspec`.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**

-------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/like_spec.rb (hunk 4 above)
Comment (Busiwa24) [reply spec/models/like_spec.rb]: resolved the `user_id` with `author_id`

File: spec/models/post_spec.rb (hunk 5 above)
Comment (Busiwa24) [reply spec/models/post_spec.rb]: resolved with change to` author_id`

**Review by Emmyn5600 — APPROVED** [review]
Hi @Busiwa24 

Woo ow 👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 

##  Optional suggestion

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**